### PR TITLE
feat(RCE): Add a touch of color to message contents

### PR DESCRIPTION
**Pull request by Kyuuhachi**
image

Adapted from my [ColorMessage](<https://github.com/Kyuuhachi/VencordPlugins/tree/main/ColorMessage>) plugin.

### CODE REVIEW

File: src/plugins/roleColorEverywhere/index.tsx [hunk 1]
```diff
@@ -46,7 +50,15 @@ const settings = definePluginSettings({
         default: true,
         description: "Show role colors in the reactors list",
         restartNeeded: true
-    }
+    },
+    messageSaturation: {
+        type: OptionType.SLIDER,
+        description: "Message color saturation",
```
Comment (Nuckyz) [inline src/plugins/roleColorEverywhere/index.tsx:56]:
```suggestion
        description: "Message color saturation. 0 to disable coloring messages",
```

File: src/plugins/roleColorEverywhere/index.tsx [hunk 2]
```diff
@@ -114,7 +126,15 @@ export default definePlugin({
                 replace: "$&,style:{color:$self.getColor($2?.id,$1)}"
             },
             predicate: () => settings.store.reactorsList,
-        }
+        },
+        {
+            find: '.Messages.MESSAGE_EDITED,")"',
+            replacement: {
+                match: /(?<=isUnsupported\]:(\i)\.isUnsupported\}\),)(?=children:\[)/,
+                replace: 'style:{"color":$self.useMessageColor($1)},'
```
Comment (Nuckyz) [inline src/plugins/roleColorEverywhere/index.tsx:134]:
```suggestion
                replace: "style:{color:$self.useMessageColor($1)},"
```

**Review by Nuckyz — CHANGES_REQUESTED** [review]

File: src/plugins/roleColorEverywhere/index.tsx [hunk 3]
```diff
@@ -175,7 +175,7 @@ export default definePlugin({
             const { messageSaturation } = settings.use(["messageSaturation"]);
             const author = useMessageAuthor(message);
             if (author.colorString !== undefined)
-                return `color-mix(in lab, ${author.colorString} calc(${messageSaturation} * 1%), var(--text-normal))`;
+                return `color-mix(in lab, ${author.colorString} ${messageSaturation}%, var(--text-normal))`;
```
Comment (lewisakura) [inline src/plugins/roleColorEverywhere/index.tsx:178]:
```suggestion
                return `color-mix(in oklab, ${author.colorString} ${messageSaturation}%, var(--text-normal))`;
```

**Review by lewisakura — CHANGES_REQUESTED** [review]

**Comment by Kyuuhachi** [comment]
Adjusted according to feedback, as well as a few other small changes.

**Comment by Nuckyz** [comment]
Thank you for the addition!

**Pull request merged**